### PR TITLE
Merge 1.7.7.x into develop - 20/03/2020

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | 
| Type?         | improvement
| Category?     | ME
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | No need for QA

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/18244)

### CODE REVIEW

**Comment by matks** [comment]
@boubkerbribri I had to fix a git conflict on file `tests/puppeteer/pages/BO/catalog/products/add.js`, can you check if I fixed it properly ?

Git wanted me to choose between `createEditProduct()` function and `setBasicSetting()` and I added both.

**Comment by boubkerbribri** [comment]
> @boubkerbribri I had to fix a git conflict on file `tests/puppeteer/pages/BO/catalog/products/add.js`, can you check if I fixed it properly ?
> 
> Git wanted me to choose between `createEditProduct()` function and `setBasicSetting()` and I added both.

Hello @matks , I couldn't make suggestion (not changed code), But you can delete all `CreateEditProduct` function.
Thank you

**Comment by matks** [comment]
Wonderful, this breaks sanity tests
```
  Create, read, update and delete Standard product in BO
    ✓ should login in BO
    ✓ should go to Products page
    ✓ should reset all filters

    1) should create Product

  21 passing (2m)
  1 failing
  1) Create, read, update and delete Standard product in BO
       should create Product:
     Error: No node found for selector: undefined
      at assert (node_modules/puppeteer/lib/helper.js:267:11)
      at DOMWorld.click (node_modules/puppeteer/lib/DOMWorld.js:366:5)
      at <anonymous>
    -- ASYNC --
      at Frame.<anonymous> (node_modules/puppeteer/lib/helper.js:111:15)
      at Page.click (node_modules/puppeteer/lib/Page.js:986:29)
      at AddProduct.setBasicSetting (pages/BO/catalog/products/add.js:60:21)
      at <anonymous>
```

File: tests/puppeteer/pages/BO/catalog/products/add.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ module.exports = class AddProduct extends BOBasePage {
     this.forNavlistItemLink = `${this.formNavList} #tab_step%ID a`;
     // Selectors of Step 2 : Pricing
     this.addSpecificPriceButton = '#js-open-create-specific-price-form';
+    this.specificPriceForm = '#specific_price_form';
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/catalog/products/add.js:36]:
```suggestion
    this.specificPriceForm = '#specific_price_form';
    this.productShortDescriptionTab = '#tab_description_short a';
    this.productDescriptionTab = '#tab_description a';
```

File: tests/puppeteer/pages/BO/catalog/products/add.js [hunk 2]
```diff
@@ -48,73 +49,67 @@ module.exports = class AddProduct extends BOBasePage {
...
-    // Set Name, type of product, Reference, price ttc and quantity, and with combinations
-    await this.page.click(this.productNameInput, {clickCount: 3});
-    await this.page.type(this.productNameInput, productData.name);
-    await this.selectByVisibleText(this.productTypeSelect, productData.type);
-    await this.page.click(this.productReferenceInput, {clickCount: 3});
-    await this.page.type(this.productReferenceInput, productData.reference);
-    await this.page.click(this.productPriceTtcInput, {clickCount: 3});
-    await this.page.type(this.productPriceTtcInput, productData.price.toString());
-    // Set description value
+  async setBasicSetting(productData) {
```
Comment (boubkerbribri) [inline tests/puppeteer/pages/BO/catalog/products/add.js:65]:
```suggestion
  /**
   * Set Name, type of product, Reference, price ttc, description and short description
   * @param productData
   * @return {Promise<void>}
   */
  async setBasicSetting(productData) {
```

**Review by boubkerbribri — COMMENTED** [review]
It seems 2 selectors were missing, don't know how, I made you suggestions to resolve the problem.
Or you can take file 1.7.7 [file](https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/tests/puppeteer/pages/BO/catalog/products/add.js) and it should be fine
Thanks @matks

**Review by matthieu-rolland — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
Thanks for review

**Pull request merged**